compiler/src/dotty/tools/dotc/typer/Typer.scala

```diff
@@ -2505,6 +2506,7 @@ class Typer(@constructorOnly nestingLevel: Int = 0) extends Namer
       // hence we special case it until `erased` is no longer experimental.
       sym.setFlag(Erased)
     val DefDef(name, paramss, tpt, _) = ddef
+    if name == nme.ROOTPKG then report.error(em"Illegal use of root package name.", ddef)
```

Comment:
Maybe factor out into a method in Checking since it is used twice (or 3 times with packages)?

Reply:
Position should be `ctx.source.atSpan(ddef.nameSpan)`. Then it's precisely the name that is highlighted.